specification/flag-evaluation/flag-evaluation.json

```
@@ -62,26 +62,26 @@
     {
         "id": "1.13",
         "clean id": "1_13",
-        "content": "The `evaluation details` structure's `variant` field SHOULD contain a identifier logically corresponding to the return flag value.",
-        "RFC 2119 keyword": "SHOULD"
+        "content": "In cases of normal execution, the `evaluation details` structure's `variant` field MUST contain the value of the `variant` field in the `flag resolution` structure returned by the configured `provider`.",
+        "RFC 2119 keyword": "MUST"
     },
     {
         "id": "1.14",
         "clean id": "1_14",
-        "content": "The `evaluation details` structure's `reason` field SHOULD indicate the semantic reason for the returned flag value. Possible values vary by provider, but may include `\"TARGETING_MATCH\"`, `\"SPLIT\"`, `\"DISABLED\"`, `\"DEFAULT\"`, `\"UNKNOWN\"` and `\"ERROR\"`.",
-        "RFC 2119 keyword": "SHOULD"
+        "content": "In cases of normal execution, the `evaluation details` structure's `reason` field MUST contain the value of the `reason` field in the `flag resolution` structure returned by the configured `provider`.",
+        "RFC 2119 keyword": "MUST"
```

Do all providers give a reason? If not, you're going to get a lot of default, or null reasons here.

---

Many do not, some do, this is an optional field. In many cases it will be missing, null, or undefined.

---

Ah, I understand now. In which case, I think this is confusing as I read 'MUST' to imply that it must return a value. Perhaps tweak the wording
```suggestion
        "content": "In cases of normal execution, the `evaluation details` structure's `reason` field MUST contain the value of the `reason` field in the `flag resolution` structure returned by the configured `provider` if the `provider` returns it (if the `provider` does not give a reason, then it does not need to be set.",
        "RFC 2119 keyword": "MUST"
```

Or some better language